flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -381,7 +381,8 @@ public static String getTaskManagerShellCommand(
 			boolean hasLogback,
 			boolean hasLog4j,
 			boolean hasKrb5,
-			Class<?> mainClass) {
+			Class<?> mainClass,
```

Comment:
I think BootstrapTools is an internal class so that we don't stick to it, but it makes sense to add an override method for preventing some compile issue. IIRC it was reported in classes like CloseClosure and CheckpointCoordinator. Also an override makes it smooth for forks to pick think commit.

Reply:
@walterddr A agree with tison that it is an internal class.
@TisonKun All the usage of `getTaskManagerShellCommand` has been updated to use the new args. So i think it is also easy to cherry-pick. If you insist, i will keep the old function.

Reply:
Make sense to do as is :-)